services/terraformer/server/domain/utils/tofu/terraform.go

```diff
@@ -37,14 +42,57 @@ type Terraform struct {
 	SpawnProcessLimit *semaphore.Weighted
 }
 
+func (t *Terraform) ProvidersLock() error {
+	absCache, err := filepath.Abs(t.CacheDir)
+	if err != nil {
+		return fmt.Errorf("failed to resolve absolute cache dir: %w", err)
+	}
+
+	if _, err := os.Stat(absCache); os.IsNotExist(err) {
+		if err := os.MkdirAll(absCache, os.ModePerm); err != nil {
+			return fmt.Errorf("failed to create cache directory %s : %w", absCache, err)
+		}
+	}
```

Comment:
you can use `fileutils.CreateDirectory` utility function for creating the cache directory.